examples/sol-contract/scripts/invoke.ts

```diff
@@ -0,0 +1,139 @@
+        ];
+
+    let initLayout = struct([ u8('instruction') ])
+    let initData = Buffer.alloc(initLayout.span);
+    let loan2ValueLayout = struct([
+        u8('instruction'), blob(8, 'loan_qty'), blob(8, 'collateral_qty')
+    ])
+    let loan2ValueData = Buffer.alloc(loan2ValueLayout.span);
+
+    /* Invoke the Init instruction (instruction #0) */
+    console.log("Creating data account and invoking Init...");
+    let txInit = new web3.Transaction({ feePayer: payer.publicKey });
+    initLayout.encode({instruction: 0}, initData);
+    txInit.add(
+        createInst,                        /* Create data account */
+        new web3.TransactionInstruction({  /* Initialize data account */
+            data: initData,
+            keys: accounts,
+            programId: contract.publicKey
+        })
+    );
+    let txInitSig = await web3.sendAndConfirmTransaction(
+        conn, txInit, [payer, dataAccount, contract]
+    );
+    console.log("TxHash: " + txInitSig);
+
+    /* Invoke the Loan2Value instruction (instruction #1) */
+    console.log("Checking loan to value ratio...");
+    /* Encode 0x1 in big ending */
+    let loan_qty = Buffer.from('0100000000000000', 'hex');
+    /* Encode 0xbb8 (3000) in big ending */
+    let collateral_qty = Buffer.from('b80b000000000000', 'hex');
+    let txCheck = new web3.Transaction({ feePayer: payer.publicKey });
+    loan2ValueLayout.encode(
+        {instruction: 1,
+         loan_qty: blob(8).decode(loan_qty),
+         collateral_qty: blob(8).decode(collateral_qty)}
+    , loan2ValueData);
+    txCheck.add(
+        new web3.TransactionInstruction({
+            data: loan2ValueData,
+            keys: accounts,
+            programId: contract.publicKey
+        })
+    );
+    let txCheckSig = await web3.sendAndConfirmTransaction(
+        conn, txCheck, [payer, contract]
+    );
+    console.log("TxHash: " + txCheckSig);
+
+    /* Try to invoke the Init instruction without authority */
+    console.log("Trying an unauthorized invocation of Init...");
+    let attacker = web3.Keypair.generate();
+    accounts[0].pubkey = attacker.publicKey
+
+    let attackerDataAccount = web3.Keypair.generate();
+    const attackerCreateInst = web3.SystemProgram.createAccount({
+        lamports: dataCost,
+        space: loanInfoSize,
+        programId: contract.publicKey,
+        fromPubkey: payer.publicKey,
+        newAccountPubkey: attackerDataAccount.publicKey,
+    });
+
+    let txAttacker = new web3.Transaction({ feePayer: payer.publicKey });
+    txAttacker.add(
+        attackerCreateInst,
+        new web3.TransactionInstruction({
+            data: initData,
+            keys: accounts,
+            programId: contract.publicKey
+        })
+    );
+
+    try {
+        let txAttackerSig = await web3.sendAndConfirmTransaction(
+            conn, txAttacker, [payer, attackerDataAccount, attacker]
+        );
+        console.error("Attacker succeeded with TxHash: " + txAttackerSig);
```

Comment:
Can you add an assert or exit after this. Because this behaviour should not happen.

Reply:
Sure, I was using `console.error` and now I changed it to `throw new Error(" ... ");`.